### DISCUSSION TITLE
Unable to login to netbox

### DISCUSSION THREAD

**Discussion by aman20010sinha** [netbox-community/netbox#16930] [Help Wanted!]
Hi,
Facing below errors while trying to open Netbox

<class 'django.db.utils.OperationalError'>

connection failed: :1), port 5432 failed: FATAL:  remaining connection slots are reserved for non-replication superuser connections

Python version: 3.11.6
NetBox version: 3.6.4
Plugins: None installed

I guess we were using username password to authenticate to netbox via API and we have reached connection limit. How to remove the API tokens from Netbox?

**Comment by markkuleinio** [comment]
This is about NetBox unable to connect to the PostgreSQL database. Search for configuration issues or unexpected database connections on the database server.

**Reply by aman20010sinha** [reply]
I restarted Netbox and the issue was resolved. I am using same token to perform multiple API operations on Netbox. Can that be the reason behind this?
I am seeing this issue more often now.

**Reply by aman20010sinha** [reply]
I am seeing this when the same token is used to make multiple API requests to netbox server. Is there any way to resolve this?

**Reply by candlerb** [reply]
This is nothing to do with tokens. It's to do with how your database is configured, in particular the number of concurrent connections it allows.

**Reply by aman20010sinha** [reply]
Okay, so in my usecase there will be multiple concurrent users who will access Netbox server. What configuration is required to allow concurrent users in netbox? Please guide me through it

**Reply by candlerb** [reply]
It works fine for the rest of us (yes, with concurrent users). Therefore either something is wrong with your database, or there's something else different about your system - e.g. you are running a custom script which is getting stuck and holding on to connections, or something like that.

You've not said anything at all about what OS distribution you're running on, what version of postgres you're running, what configuration changes you've made (if any), etc etc. Maybe you're using an odd OS distribution which doesn't have a sensible set of defaults for postgres.

`ss -ntp` will show you all the open TCP sockets, and you need to start debugging from there - postgres is usually port 5432. What connection limit do you hit? What's causing the number of connections to increase?

Sorry, but there's no magic wand we can wave. You need to investigate. Good luck.
